rosetta/retriever/retriever.go

```diff
@@ -231,30 +231,44 @@ func (r *Retriever) Block(id identifier.Block) (*object.Block, []identifier.Tran
 		count++
 	}
 
-	// We need the *uint64 for the parent block identifier.
-	var parentHeight *uint64
-	if header.Height > 0 {
-		h := header.Height - 1
-		parentHeight = &h
-	}
-
 	// Now we just need to build the block.
 	block := object.Block{
 		ID: identifier.Block{
 			Index: &header.Height,
 			Hash:  header.ID().String(),
 		},
-		ParentID: identifier.Block{
-			Index: parentHeight,
-			Hash:  header.ParentID.String(),
-		},
+		ParentID:     parentBlock(header),
 		Timestamp:    header.Timestamp.UnixNano() / 1_000_000,
 		Transactions: blockTransactions,
 	}
 
 	return &block, extraTransactions, nil
 }
 
+func parentBlock(header *flow.Header) identifier.Block {
```

Comment:
I'm really not sure about the value of creating a function for this. We can just have this in the body of the retriever method like this:

```
id := identifier.Block{ ... }
parent := id
if header.Height > 0 {
  parent = identifier.Block { ... }
}
```
```

Reply:
Yeah, initially I felt it was a smidge more readable. Now with the flow restructured, it looks pretty irrelevant.